src/transformers/generation/utils.py

```diff
@@ -1497,6 +1482,127 @@ def _supports_default_dynamic_cache(self) -> bool:
         """
         return self._supports_cache_class and "jamba" not in self.__class__.__name__.lower()
 
+    def _prepare_cache_for_generation(
+        self,
+        generation_config: GenerationConfig,
+        model_kwargs: Dict,
+        assistant_model: "PreTrainedModel",
+        batch_size: int,
+        device: torch.device,
+    ) -> bool:
+        """
+        Prepares the cache for generation (if applicable), given `generate`'s paramaterization. If a cache is
+        instantiated, writes it to `model_kwargs`, under the name expected by the model.
+        """
+
+        cache_name = "past_key_values" if "mamba" not in self.__class__.__name__.lower() else "cache_params"
+        requires_cross_attention_cache = (
+            self.config.is_encoder_decoder or model_kwargs.get("encoder_outputs") is not None
+        )
+
+        # Quick escape route 1: if the user specifies a cache, we only need to:
+        # a) check for conflicting `generate` arguments
+        # b) convert to the new cache format (if the user passes a legacy cache and model supports it)
+        user_defined_cache = model_kwargs.get(cache_name)
+        if user_defined_cache is not None:
+            if is_torchdynamo_compiling():
+                raise ValueError(
+                    "Passing `past_key_values` is not supported when compiling `model.generate` with torch.compile -- "
+                    "you may get incorrect outputs. Please compile `model.forward` only or use the "
+                    "`cache_implementation` input argument."
```

Comment:
Unless the cache is static no?

Reply:
It wasn't working in the PR that added this exception, but I've checked again due to your comment and it seems to be working. Possibly because it now inherits from `nn.module`? 🤷 

Anyways, removing the exception 🤗